lib/immobilienscout/request.rb

```diff
@@ -48,6 +48,19 @@ def post_with_multipart
       execute_request(uri, request)
     end
 
+    def put
+      auth_header = generate_auth_header('PUT')
+      headers = generate_headers(auth_header)
+
+      uri = URI.parse(@url)
+
+      request = Net::HTTP::Put.new(uri)
+      request.body = @params.to_json
+      request.initialize_http_header(headers)
+
+      execute_request(uri, request)
+    end
```

Comment:
I feel like we are repeating the same code in each action. But not hundred percent, I know 😅 - but maybe we can find a way to generalize at some point.

Reply:
It would be a big refactor work! 😄

Reply:
Yes, that's why I added the "at some point"... 🙈